controllers/controllers/resource/reconciler.go

```diff
@@ -96,6 +103,42 @@ func (cor *clusterReconciler) Reconcile(ctx context.Context, objectKey types.Nam
 			return err
 		}
 		resources = append(resources, r...)
+	case anywherev1.CloudStackDatacenterKind:
+		if !features.IsActive(features.CloudStackProvider()) {
+			return fmt.Errorf("cloudstack provider is not supported in eks-a controller")
+		}
+		csdc := &anywherev1.CloudStackDatacenterConfig{}
+		cpCsmc := &anywherev1.CloudStackMachineConfig{}
+		etcdCsmc := &anywherev1.CloudStackMachineConfig{}
+		workerCsmc := &anywherev1.CloudStackMachineConfig{}
+		workerCsmcs := make(map[string]anywherev1.CloudStackMachineConfig, len(cs.Spec.WorkerNodeGroupConfigurations))
+		err := cor.FetchObject(ctx, types.NamespacedName{Namespace: objectKey.Namespace, Name: cs.Spec.DatacenterRef.Name}, csdc)
+		if err != nil {
+			return err
+		}
+		err = cor.FetchObject(ctx, types.NamespacedName{Namespace: objectKey.Namespace, Name: cs.Spec.ControlPlaneConfiguration.MachineGroupRef.Name}, cpCsmc)
+		if err != nil {
+			return err
+		}
+		if len(cs.Spec.WorkerNodeGroupConfigurations) != 1 {
```

Comment:
Is this not something that should also be a part of the validating webhook?

Reply:
Do you mean on create webhook? Upgrade?

Reply:
Whatever it applies to? Forgive me, I'm not 100% how we're using webhooks at present but based on other comments we already have a validating webhook so I'm not quite tracking why this wouldn't be there?

Reply:
We can probably read the DatacenterRef to see if CloudStack is specified and then check based on that and the feature flag here.

Reply:
Done